### PR TITLE
PPP and multiple fixes for UGDG

### PR DESCRIPTION
**Pull request by fsgmhoward**
- Reverted changes for handling of NoSuchMethodException in Parser
- PPP for @fsgmhoward
- Fix problems with Table of Content in DG
- Fix typos and format problems in both UG and DG
- Update DG on Exception and common classes
- Remove unused import for RetrieveCommand

### CODE REVIEW

**Review by brandonfoong — APPROVED** [review]
LGTM 👍

File: docs/team/fsgmhoward.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# ZH Liu — Project Portfolio Page
+
+## Overview
+Patient Manager is a _Command Line Interface_ (CLI) application for **_general practitioners_** (GP)
+who work in polyclinics to manage their patient list.
+
+My teammate, together with me, developed this application as a team project for CS2113T module of NUS.
```
Comment (brandonfoong) [inline docs/team/fsgmhoward.md:7]: Minor grammar check here - should be "I, together with my teammates..."

File: docs/team/fsgmhoward.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+This project aimed to develop a program targetting a current problem using Java with Object-oriented
```
Comment (brandonfoong) [inline docs/team/fsgmhoward.md:8]: Typo here, should be "targeting"

File: docs/team/fsgmhoward.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+programming as model.
+
+The entire development cycle, excluding the planning stage which consists of a few weeks before actual
+development, includes three iterations. V1.0 is the minimum viable product, and V2.0 is the one with most
+of the features in. For v2.1, the last iteration, we focused on bug fixing and improvements in product
```
Comment (brandonfoong) [inline docs/team/fsgmhoward.md:13]: For consistency, should all version numbers start with a small letter v?

File: docs/team/fsgmhoward.md (hunk 1 above)
Comment (fsgmhoward) [reply docs/team/fsgmhoward.md]: Thank you! This has been resolved in the latest commit.

File: docs/team/fsgmhoward.md (hunk 3 above)
Comment (fsgmhoward) [reply docs/team/fsgmhoward.md]: Thank you! This has been resolved in the latest commit.

File: docs/team/fsgmhoward.md (hunk 2 above)
Comment (fsgmhoward) [reply docs/team/fsgmhoward.md]: Thank you! This has been resolved in the latest commit. My local proof reader didn't report it as a typo. It probably understood it as British spelling. After checking articles on this, I guess you are correct.

**Pull request merged**